app/ui-react/packages/ui/src/Customization/apiClientConnectors/create/securityValidation.ts

```diff
@@ -0,0 +1,27 @@
+import { ICreateConnectorPropsUi } from './ApiConnectorCreatorSecurityForm';
+
+export interface IErrorValidation {
+  password?: string;
+  username?: string;
+}
+
+export default function validateSecurity(values: ICreateConnectorPropsUi) {
+  const errors: IErrorValidation = {};
+
+  if (
+    values.authenticationType === 'basic' ||
+    values.authenticationType === 'ws-security-ut'
+  ) {
+    if (values.passwordType !== 'PasswordNone') {
+      if (!values.username) {
+        errors.username = 'Username is required';
```

Comment:
How would you localize this string, is it shown to the user?

Reply:
Good point. Yes, this does get shown to the user. Actually it's not such a large file that I can't just put it in the existing component, or I can create a separate component for it entirely?